pkg/listwatch/listwatch_test.go

```diff
@@ -458,30 +521,37 @@ func TestMultiWatchDoesNotDeadlock(t *testing.T) {
 			},
 			WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
 				if namespace == "4" {
-					return nil, errors.New("some error occurred")
+					defer func() { errorRaised <- struct{}{} }()
+					return nil, errors.New("deliberate error forced")
 				}
+				wsLock.Lock()
+				defer wsLock.Unlock()
 				w := watch.NewFake()
 				ws[namespace] = w
 				return w, nil
 			},
 		}
-	})
+	}, &v1.Pod{}, 0)
 
 	if _, err := m.List(metav1.ListOptions{}); err != nil {
 		t.Errorf("Error while listing: %v", err)
 	}
 
-	if _, err := m.Watch(metav1.ListOptions{}); err == nil {
-		t.Error("Expected error while watching, got nil")
+	<-errorRaised
+	<-errorRaised
```

Comment:
why wait for two errors to be raised?

Reply:
It waits for two errors to be raised so we can ensure that the watch is being retried, I can add a comment about it.